src/t8_forest/t8_forest_vtk.cxx

```diff
@@ -306,14 +358,67 @@ t8_forest_write_vtk_via_API (t8_forest_t forest, const char *fileprefix)
  * 
  * Note: We could write more than one file per process here, if desired.
  */
-  pwriterObj->SetNumberOfPieces (forest->mpisize);
-  pwriterObj->SetStartPiece (forest->mpirank);
-  pwriterObj->SetEndPiece (forest->mpirank);
-  pwriterObj->SetInputData (unstructuredGrid);
-  pwriterObj->Update ();
-  pwriterObj->Write ();
-/* We have to free the allocated memory for the cellTypes Array. */
-  T8_FREE (cellTypes);
+pwriterObj->SetNumberOfPieces (forest->mpisize);
+pwriterObj->SetStartPiece (forest->mpirank);
+pwriterObj->SetEndPiece (forest->mpirank);
+  /* *INDENT-OFF* */
+  if (write_treeid == 1) {
```

Comment:
You can remove the ==1 part here, and in the other ifs.
This also has the benefit of letting the user provide any non-zero value (which we usually interpret as true)

```suggestion
  if (write_treeid) {
```